AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter+SignedInSession.swift

```diff
@@ -0,0 +1,194 @@
+            guard let self = self else {
+                return
+            }
+            guard error == nil else {
+
+                if let urlError = error as NSError?, urlError.domain == NSURLErrorDomain {
+                    let authSession = AuthCognitoSignedInSessionHelper.offlineSignedInSession()
+                    completionHandler(.success(authSession))
+
+                } else if let awsMobileClientError = error as? AWSMobileClientError {
+                    if case .unableToSignIn = awsMobileClientError {
+                        // session expired.
+                        let authSession = AuthCognitoSignedInSessionHelper.expiredSignedInSession()
+                        completionHandler(.success(authSession))
+                    }
+                } else if let cognitoIdentityPoolError = error as NSError?,
+                    cognitoIdentityPoolError.domain == AWSCognitoIdentityErrorDomain,
+                    cognitoIdentityPoolError.code == AWSCognitoIdentityErrorType.notAuthorized.rawValue {
+                    // Cognito Identity Pool is not configured for this authentication provider.
+
+                    let identityIdError = AmplifyAuthError.service(
+                        AuthPluginErrorConstants.signedInIdentityIdWithNoCIDPError.errorDescription,
+                        AuthPluginErrorConstants.signedInIdentityIdWithNoCIDPError.recoverySuggestion,
+                        AWSCognitoAuthError.invalidAccountTypeException)
+                    let awsCredentialsError = AmplifyAuthError.service(
+                        AuthPluginErrorConstants.signedInAWSCredentialsWithNoCIDPError.errorDescription,
+                        AuthPluginErrorConstants.signedInAWSCredentialsWithNoCIDPError.recoverySuggestion,
+                        AWSCognitoAuthError.invalidAccountTypeException)
+                    let authSession = AWSAuthCognitoSession(isSignedIn: true,
+                                                            userSubResult: userSubResult,
+                                                            identityIdResult: .failure(identityIdError),
+                                                            awsCredentialsResult: .failure(awsCredentialsError),
+                                                            cognitoTokensResult: tokenResult)
+                    completionHandler(.success(authSession))
+                }
+                let authError = AuthErrorHelper.toAmplifyAuthError(error!)
+                let authSession = AuthCognitoSignedInSessionHelper.signedInSession(withUnhandledError: authError)
+                completionHandler(.success(authSession))
+                return
+            }
+
+            guard let credentials = awsCredentials else {
+                // This should not happen, return an unknown error.
+                let error = AmplifyAuthError.unknown("""
+                    Could not fetch the AWS Credentials
```

Comment:
as above, be explicit about the expectations -- why we believe this shouldn't happen